packages/std/src/metadata.rs

```diff
@@ -0,0 +1,23 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::math::Uint128;
+
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, JsonSchema)]
+pub struct DenomMetadata {
+    pub description: String,
+    pub denom_units: Vec<DenomUnit>,
+    pub base: String,
+    pub display: String,
+    pub name: String,
+    pub symbol: String,
+    pub uri: String,
+    pub uri_hash: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, JsonSchema)]
+pub struct DenomUnit {
+    pub denom: String,
+    pub exponent: Uint128,
```

Comment:
Can we use a smaller type? The original type is uint32.

Reply:
yes, I was looking at the `math` crate, which doesn't have a `Uint32` type, but I'll change this to use a primitive `u32`.

Reply:
Uint64/Uint128/Uint256/Uint512 to support numbers > 53 bit over JSON without a loss. They are enoded as strings. u32/i32 become simple JSON numbers and are safe to use in the full 32bit range.